client/hwlib/src/models/request_validators.rs

```diff
@@ -129,7 +130,9 @@ impl CertificationStatusRequest {
         let model = system_info.product_name;
         let vendor = system_info.manufacturer;
 
-        let architecture = get_architecture(runner)?;
+        let architecture = to_debian_architecture(ARCH)
+            .with_context(|| format!("cannot parse architecture: {ARCH:?}"))?
```

Comment:
I think we should be careful about the use of colons in error messages, here. I think after a colon, the  reader can expect to see some inner reason for the error, but here, it's just the data which caused the error. Perhaps it would  be cleaner to drop the colon after architecture? If this is an existing convention across the codebase, feel free to address this in another PR